ush/python/pygfs/task/stat_analysis.py

```diff
@@ -0,0 +1,185 @@
+        Parameters
+        ----------
+        None
+        Returns
+        ----------
+        None
+        """
+        # Create dictionary of Jedi objects
+        # Expected keys are what must be included from the JEDI config file. We can
+        # then loop through ob space list from scripts/exglobal_analysis_stats.py
+        expected_keys = ['aero', 'snow']
+        self.jedi_dict = Jedi.get_jedi_dict(self.task_config.JEDI_CONFIG_YAML, self.task_config, expected_keys)
+
+        logger.info(f"Copying files to {self.task_config.DATA}/stats")
+
+        # Loop through ob space list
+        for OB in self.task_config.STAT_OBS:
+            logger.info(f"Working on current observation: {OB}")
+
+            # Parse JEDI analysis stat jinja file
+            obs_dict = parse_j2yaml(self.task_config.JEDI_CONFIG_YAML, self.task_config)
+
+            # Copy stat files to DATA path
+            instat_files = os.path.join(obs_dict[OB]['stat_file_path'], f"{self.task_config['APREFIX']}{obs_dict[OB]['stat_file_name']}")
+            ob_dir_str = f"{self.task_config.DATA}" + f"/{OB}"
+            os.mkdir(ob_dir_str)
+
+            dest = os.path.join(ob_dir_str, obs_dict[OB]['stat_file_name'])
+            logger.info(f"Copying {instat_files} to {dest} ...")
+            statlist = [[instat_files, dest]]
+            FileHandler({'copy': statlist}).sync()
+
+            # Open tar file
+            logger.info(f"Open tarred stat file in {dest}")
+            with tarfile.open(dest, "r") as tar:
+                # Extract all files to the current directory
+                tar.extractall(path=f'{ob_dir_str}')
+
+            # Gunzip .nc files
+            logger.info("Gunzip files from tar file")
+            gz_files = glob.glob(os.path.join(ob_dir_str, "*.gz"))
+            logger.info(f"Gunzip files: {gz_files}")
+
+            for diagfile in gz_files:
+                output_file = os.path.join(ob_dir_str, os.path.basename(diagfile)[:-3])
+                with gzip.open(diagfile, 'rb') as f_in:
+                    with open(output_file, 'wb') as f_out:
+                        f_out.write(f_in.read())
+
+            # Get list of .nc4 files
+            # obs_space_paths = glob.glob(os.path.join(ob_dir_str, "*.{nc,nc4}")) # THIS SHOULD WORK BUT ISNT, glob patterns introduced in Python 3.9
+            nc_paths = glob.glob(os.path.join(ob_dir_str, "*.nc"))
+            nc4_paths = glob.glob(os.path.join(ob_dir_str, "*.nc4"))
+            obs_space_paths = nc_paths + nc4_paths
+
+            # Temporary. Create condition check here for available jcb algorithms?
+            if OB == 'snow':
+                obs_space_paths = glob.glob(os.path.join(ob_dir_str, "diag_ims_snow_*.nc"))
+
+            # This grabs the obspace string from the .nc4 files, however not all are perfect. Need solution.
+            self.task_config.OBSPACES_LIST = ['_'.join(os.path.basename(path).split('_')[1:3]) for path in obs_space_paths]
```

Comment:
Cory and I were discussing this offline. Maybe would be best to have some sort of configuration file to avoid grabbing the wrong files. Also not all file names follow the same format mentioned in comment on line 133. This would remove the temporary fix on lines 130-131.

Reply:
Yes this is too rigid and prone to failure, this and the above globs are risky. We should have a list of files that are expected based on some configuration file